components/dashboards-web-component/src/utils/DashboardUtils.jsx

```diff
@@ -100,4 +101,76 @@ export default class DashboardUtils {
         return s4() + s4() + '-' + s4() + '-' + s4() + '-' +
             s4() + '-' + s4() + s4() + s4();
     };
+
+    /**
+     * helper function to traverse the content of a page and find widgets to extract their options to a
+     * given map
+     * @param page : the page to be traversed
+     * @param OptionsMap : the map to extract option configurations
+     * */
+    static searchForWidgetsInAPage(page, OptionsMap) {
+        let obj = page;
+        if (obj.type && obj.type === 'component' && obj.props && obj.props.id) {
```

Comment:
Please change this as
`if (obj.type && obj.props && obj.props.id && obj.type === 'component') {`

Reply:
the conditional statements in javascript are evaluated from Left to Right refer to Short-circuit evaluation. I only want to check the props if the object is of type component. The code is written according to that logic so let's keep it that way.

Reply:
&& is used here. That means, **every** condition should be true. If at least one of them is false, we are not going to check for the next ones. If you only want to check whether the obj.type === 'component', then you would have had it as `obj.type && obj.type === 'component'`.
We are making it to get out of the condition as quick as it can, where there is a probability for the condition to become `false`.

**Please use this logic in the other places I have mentioned like this too. Same rationale applies for them**

Reply:
Let me jump on this argument :smiley: 

Why can't we just use `obj.type === 'component'`?? 
`obj.type && obj.type === 'component'` equals to `obj.type != null && obj.type === 'component'`, so when `obj.type === 'component'` then automatically `obj.type != null`, right.

Reply:
`obj.type && obj.type === 'component'`
The first condition is for checking whether obj.type has been defined or not. There can be `undefined` scenarios where we get things like 'Can't read 'id' of undefined'.

Refer this:

![image](https://user-images.githubusercontent.com/24828296/37818911-bc4fa12a-2ea1-11e8-94d1-f7366b0b6d13.png)

![image](https://user-images.githubusercontent.com/24828296/37818948-d733db0a-2ea1-11e8-80c6-f6e1be7bec95.png)


![image](https://user-images.githubusercontent.com/24828296/37818967-e630c15e-2ea1-11e8-9b30-e0e6059452ec.png)

So we have to check whether we have `type` and `props` defined first of all, then access them and get their members.

Reply:
Yes, then we can use as:
`obj.type === 'component' && obj.props && obj.props.id`
No need to check `obj.type` first